Add Pay By Link

gcatanese opened the pull request:
Add Pay By Link use case

jlengrand commented:
* The link will not be created if no reference is entered, is it per design? Wasn't like this is in my implementation.
* The form does not reset itself once a link is created

gcatanese commented:
@jlengrand Thanks for the feedback, I have made some changes:

`The link will not be created if no reference is entered, is it per design? Wasn't like this is in my implementation.`
Reference appears to be mandatory (API Explorer) so we don't create the link (no form submission)
We could implement a better client-side validation (*)

`The form does not reset itself once a link is created`
It should be ok now

`I see alignment issues, but that can be fixed later`
Fixed

`reload button`: at the moment we refresh the list every time we reload the home page (ie navigating back after a payment). The reload button is a good idea, maybe we can re-introduce it (*)

`isReusable empty`: it was a bug and it is now fixed

(*) for those improvements I would create a different issue and plan them in the next iteration

jlengrand commented:
Created DR-258 and DR-259 for the client side validation and the reload button respectively
